app_core.py

```diff
@@ -9,8 +9,27 @@
 from addresswatcher import AddressWatcher
 from timer import Timer
 
+class ReverseProxied(object):
+    """
+    Because we are reverse proxied from an aws load balancer
+    use environ/config to signal https
+    since flask ignores preferred_url_scheme in url_for calls
+    """
+
+    def __init__(self, app):
+        self.app = app
+
+    def __call__(self, environ, start_response):
+        # if one of x_forwarded or preferred_url is https, prefer it.
+        forwarded_scheme = environ.get("HTTP_X_FORWARDED_PROTO", None)
```

Comment:
does it crash is HTTP_X_FORWARDED_PROTO or PREFERRED_URL_SCHEME are not present in the envrionment?

Reply:
changed the variable to PREFERRED_URL_SCHEME_2 and it seems to work.